### DISCUSSION TITLE
The element index found by the iOS class chain is different between xcuitest@4.12.2 and xcuitest@4.33.0

### DISCUSSION THREAD

**Discussion by klauslaiRC** [appium/appium#18983] [General]
Env:
Appium server version (output of appium --version): 2.0.1
Appium driver(s) and their version(s): xcuitest@4.33.0 or xcuitest@4.12.2
Appium plugin(s) and their version(s):
images@2.1.2
relaxed-caps@1.0.
Node.js version (output of node --version): v18.17.0
npm version (output of npm --version): 9.6.7
Last component(s) version which did not exhibit the problem:
Platform and version under test:
MacOS Ventura 13.4.1
Xcode 15 beta 5

Question:
There are 2 elements in app page:
![image](https://github.com/appium/appium/assets/105275722/d25d196e-bea1-4dc0-aae5-dc42c9acaff4)

if i search element by "**/XCUIElementTypeNavigationBar[1]", i will get upper XCUIElementTypeNavigationBar with xcuitest@4.12.2, and get lower XCUIElementTypeNavigationBar with xcuitest@4.33.0

**Comment by jlipps** [comment]
This is one reason you should never use purely hierarchical element queries. Your app could have different structure from version to version. WDA could have different structure. XCUITest could produce different structure. In general there is no guarantee that app xml structure will be constant from release to release.

**Comment by klauslaiRC** [comment]
Maybe WDA have different structure, thanks for @jlipps
